.circleci/config.yml

```diff
@@ -156,29 +156,42 @@ jobs:
       - checkout
       - attach_workspace:
           at: .
+      - restore_cache:
+          keys:
+            - nix-v1-{{ checksum "nix/nixpkgs.json" }}
       - run:
           name: build
           command: |
             set -ex
             sudo mkdir -p /nix
-            sudo docker run --rm --privileged -ti -v /:/mnt nixos/nix cp -rfT /nix /mnt/nix
+            sudo mkdir -p .cache
+            sudo mount --bind .cache /nix
+            if [[ -z $(ls -A /nix) ]]; then sudo docker run --rm --privileged -ti -v /:/mnt nixos/nix cp -rfT /nix /mnt/nix; fi
```

Comment:
I'm still wondering why we cannot get rid of this docker execution wrapped in a VM? Using the nixos/nix container image directly here seems way more straight forward from my point of view.

Reply:
1. Basically because not all CI service like container with non-bash based. `nixos/nix` only come with `/bin/sh` so when I tried with GitHub Action/Travis CI/Cirrus CI/Circle CI across containers/* projects, some of them (as I remember should be Cirrus CI which used by conmon/buildah/podman) are not happy with ;-(
2. Another use case is crun: it is now wrapping the build of static binary into its testing script, and the script itself already running inside VM. Migrate to Docker based require for a rewrite of CI logic :-(
3. The cache folder also need to be handle in care, e.g. Cirrus CI having bug with it (see https://github.com/cirruslabs/cirrus-ci-docs/issues/681#issuecomment-665410656). Using VM based could provide more flexibility for our implementation.

P.S. If we could simplify the choose of CI platform across containers/* and cri-o/cri-o, and that selected CI platform support Docker based test for nix build simply (+ cache), I would vote for Docker based for sure ;-)